util/git/client.go

```diff
@@ -421,10 +421,16 @@ func (m *nativeGitClient) Fetch(revision string) error {
 func (m *nativeGitClient) LsFiles(path string, enableNewGitFileGlobbing bool) ([]string, error) {
 	if enableNewGitFileGlobbing {
 		// This is the new way with safer globbing
-		err := os.Chdir(m.root)
+		cwd := m.root
+		err := os.Chdir(cwd)
 		if err != nil {
 			return nil, err
 		}
+		cwd, err = os.Getwd()
```

Comment:
```suggestion
		// Depending on the OS, this can be different from m.root. Get the OS-specific path
		// for cross-platform compatibility.
		cwd, err = os.Getwd()
```

Reply:
Are they different because `os.Getwd()` is indeterministic when a folder can be reached by multiple paths via symlinks?